src/components/DopplerPlus/PlanCalculator.js

```diff
@@ -48,48 +92,75 @@ const PlanCalculator = () => {
               ¿Cuántos contactos tienes? Utiliza el slider para calcular el costo final de tu Plan
             </p>
             <div style={{ marginBottom: '40px' }}>
-              {discountsList.map((discount, index) => (
+              {state.discountsList.map((discount, index) => (
                 <button
                   key={index}
                   style={{
                     padding: '10px',
                     border: '1px solid #000',
-                    backgroundColor: discount.id === currentDiscount.id ? '#33ad73' : '#f6f6f6',
+                    backgroundColor: discount.id === planData.discount.id ? '#33ad73' : '#f6f6f6',
                   }}
                   onClick={() => {
-                    applyDiscount(index);
+                    dispatchPlanData({
+                      type: actionTypes.UPDATE_SELECTED_DISCOUNT,
+                      idDiscount: discount.id,
+                    });
                   }}
                 >
                   {discount.description}
                 </button>
               ))}
             </div>
-            {currentDiscount.percent ? (
+            {planData.discount.percent ? (
               <p style={{ textDecoration: 'line-through' }}>
-                US${currentPlan.price * currentDiscount.monthsAmmount}
+                US${planData.plan.price * planData.discount.monthsAmmount}
               </p>
             ) : (
               <></>
             )}
             <span>US$</span>
             <span style={{ fontSize: '40px' }}>
               {Math.round(
-                currentPlan.price *
-                  (1 - currentDiscount.percent / 100) *
-                  currentDiscount.monthsAmmount,
+                planData.plan.price *
+                  (1 - planData.discount.percent / 100) *
+                  planData.discount.monthsAmmount,
```

Comment:
Maybe these (and other) calculous could be done in the reducer, to simplify the template.

Reply:
I'm not sure, but it seems to be that it's not too easy as appears. 🤔

Reply:
Ok, by the moment it is fine, we can refactor it later if we consider that it could be useful.